pkg/cloud/vsphere/vsphere_config_transformer_test.go

```diff
@@ -0,0 +1,355 @@
+			infraBuilder:     newVsphereInfraBuilder().withVSphereZones(),
+			inputConfig:      iniConfigWithExistingLabels,
+			equivalentConfig: iniConfigZonal,
+		},
+		{
+			name:             "yaml and ini config parsing results should be the same",
+			infraBuilder:     newVsphereInfraBuilder(),
+			inputConfig:      yamlConfig,
+			equivalentConfig: iniConfigWithoutWorkspace,
+		},
+		{
+			name:             "yaml and ini config parsing results should be the same, with zones",
+			infraBuilder:     newVsphereInfraBuilder().withVSphereZones(),
+			inputConfig:      yamlConfigZonal,
+			equivalentConfig: iniConfigZonal,
+		},
+		{
+			name:             "yaml and ini config parsing results should be the same, node networking",
+			infraBuilder:     newVsphereInfraBuilder().withVSphereNodeNetworking(),
+			inputConfig:      yamlConfigNodeNetworking,
+			equivalentConfig: iniConfigNodeNetworking,
+		},
+		{
+			name:             "yaml config should contain node networking if it's specified in infra",
+			infraBuilder:     newVsphereInfraBuilder().withVSphereNodeNetworking(),
+			inputConfig:      yamlConfig,
+			equivalentConfig: yamlConfigNodeNetworking,
+		},
+		{
+			name:             "yaml config should be populated with datacenters and labels if failure domains specified",
+			infraBuilder:     newVsphereInfraBuilder().withVSphereZones(),
+			inputConfig:      yamlConfig,
+			equivalentConfig: yamlConfigZonal,
+		},
+		{
+			name:         "empty input",
+			infraBuilder: newVsphereInfraBuilder(),
+			errMsg:       "failed to read the cloud.conf: vSphere config is empty",
+		},
+		{
+			name:         "incorrect platform",
+			infraBuilder: newVsphereInfraBuilder().withPlatform(configv1.NonePlatformType),
+			errMsg:       "invalid platform, expected to be VSphere",
+		},
+		{
+			name:         "invalid ini input",
+			infraBuilder: newVsphereInfraBuilder(),
+			inputConfig:  ":",
+			errMsg:       "failed to read the cloud.conf",
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			g := gmg.NewWithT(t)
+			infraResouce := tc.infraBuilder.Build()
+			transformedConfig, err := CloudConfigTransformer(tc.inputConfig, infraResouce, makeDummyNetworkConfig())
+			if tc.errMsg != "" {
+				g.Expect(err).Should(gmg.HaveOccurred())
+				g.Expect(err.Error()).Should(gmg.ContainSubstring(tc.errMsg))
+				return
+			}
+
+			// Using CPI config reader from cloud-provider-vsphere
+			// to ensure that config transformation produces valid yaml config which
+			// will be readable and usable by the CCM then
+
+			wantConfig, err := ccm.ReadCPIConfig([]byte(tc.equivalentConfig))
+			if err != nil {
+				g.Expect(err).Should(gmg.Equal(nil))
+				return
+			}
+			gotConfig, err := ccm.ReadCPIConfig([]byte(transformedConfig))
+			if err != nil {
+				g.Expect(err).Should(gmg.Equal(nil))
+				return
+			}
+
+			g.Expect(gotConfig).Should(gmg.BeComparableTo(wantConfig))
```

Comment:
Why `BeComparableTo` vs `Equal` here?

Reply:
My main reasoning was a nice diff which`BeComparableTo` spits out.

Speaking about differences - under the hood `BeComparableTo` uses [go-cmp](https://pkg.go.dev/github.com/google/go-cmp/cmp#Equal) which seems to be ok alternative to `DeepEqual` for tests.